### ISSUE TITLE
Enable SSL for Telit DE910

### ISSUE THREAD

**Issue by mdbartos** [open-storm/perfect-cell#2]

**Comment by mdbartos** [comment]
Looks like the DE910 doesn’t work with some cipher suites. From the Telit SSL/TLS guide:

```
If the server involved in the connection doesn't support one of these cipher suites, the
protocol handshake fails.

Warning – the product series HE920 / UE910 V2/ DE910 do not support
TLS_RSA_WITH_NULL_SHA and TLS_RSA_WITH_AES_256_CBC_SHA
```
Xively tries to use TLS_RSA_WITH_AES_256_CBC_SHA by default

**Comment by mdbartos** [comment]
Supported cipher suites for Telit DE910 include:

```
1 - TLS_RSA_WITH_RC4_128_MD5
2 - TLS_RSA_WITH_RC4_128_SHA
3 - TLS_RSA_WITH_AES_128_CBC_SHA
```

**Comment by mdbartos** [comment]
After a lot of trial and error, @imondrag and I figured it out: the server certificate has to be signed with SHA1 instead of SHA2/SHA256. The DE910 module only supports certificates signed with SHA1. Most servers nowadays use SHA2, which is why the module refuses to connect to xively.

You can use sslscan to check the SSL certificate signing algorithm:

```
sslscan api.xively.com:443

...

  SSL Certificate:
Signature Algorithm: sha256WithRSAEncryption
RSA Key Strength:    2048
```

In order for the DE910 to make a secure connection, the signature algorithm has to read `Signature Algorithm: sha1WithRSAEncryption`.

**Comment by mdbartos** [comment]
Secured connection from DE910 to Influxdb confirmed to be working with server authentication mode (still need to test server/client authentication mode).

```
import serial
import time
import random
from time import sleep

# Modem config
modem_port = serial.Serial()
modem_port.baudrate = 115200
#modem_port.port = "/dev/tty.USA19H141P1.1" #change for different computers
modem_port.port = "/dev/tty.usbserial-FT97QV8X"
modem_port.timeout = 2 #increase timeout in case of bad reception
modem_port.open()

def modem_write(command, add_r=True):
    if isinstance(command, str):
        if add_r:
            command = command + '\r'
        command = (command).encode()
    else:
        raise TypeError('command must be a string')
    modem_port.write(command)

def send_at_command(command,timeout=2):
    modem_port.timeout = timeout
    modem_write(command)
    response = modem_port.read(3000)
    print(response)
    return response

# Check registration
creg = send_at_command('AT+CREG?')

with open('/Users/mdbartos/Documents/auth/mdbartos_selfsigned_cert.pem') as cafile:
    mdbartos_cert = (''.join(cafile.readlines())).strip() + '\n'

influx_get_header = \
"GET /query?db=_TEST&q=SELECT%20last(value)%20from%20test_measurement HTTP/1.1\r\n"+\
"Host: ec2-52-87-156-130.compute-1.amazonaws.com:8086\r\n"+\
"Connection: Close\r\n\r\n\032"

# Check if context and ssl have already been enabled
sgact = send_at_command('AT#SGACT?')
sgact = int(sgact.split(b'\r\n')[1].split(b',')[-1])
sslen = send_at_command('AT#SSLEN?')
sslen = int(sslen.split(b'\r\n')[1].split(b',')[-1])
if not sgact:
    # Activate context
    send_at_command('AT#SGACT=1,1') 
if not sslen:
    # Enable secure socket
    send_at_command('AT#SSLEN=1,1')

send_at_command('AT#SSLSECDATA=1,0,1') # Delete CA certificate
send_at_command('AT#SSLSECDATA=1,1,1,{0}'.format(len(mdbartos_cert))) # Store CA certificate
modem_write(mdbartos_cert+'\032')
send_at_command('')

# open the socket and send requests
send_at_command('AT#SSLCFG=1,1,1000,90,100,50')
send_at_command('AT#SSLSECCFG=1,0,1,1')  # Set server authentication mode
send_at_command('AT#SSLD=1,8086,"ec2-52-87-156-130.compute-1.amazonaws.com",0,1') # Wait 15s for CONNECT (LED comes back on once connection is made)
sleep(15)
# Check socket status
send_at_command('AT#SSLS=1')
# Send data
send_at_command('AT#SSLSEND=1')
send_at_command(influx_get_header)
sleep(5)
send_at_command('AT#SSLRECV=1,1000')
# Close ssl socket
send_at_command('AT#SSLH=1')
```

Output:

```
b'AT#SSLD=1,8086,"ec2-52-87-156-130.compute-1.amazonaws.com",0,1\r\r\nOK\r\n'

b'AT#SSLS=1\r\r\n#SSLS: 1,2,3\r\n\r\nOK\r\n'

b'AT#SSLSEND=1\r\r\n> '

b'GET /query?db=_TEST&q=SELECT%20last(value)%20from%20test_measurement HTTP/1.1\r\nHost: ec2-52-87-156-130.compute-1.amazonaws.com:8086\r\nConnection: Close\r\n\x1a\r\n\r\nOK\r\n\r\nSSLSRING: 1,212\r\n'

b'AT#SSLRECV=1,1000\r\r\n#SSLRECV: 370\r\nHTTP/1.1 200 OK\r\nConnection: close\r\nContent-Type: application/json\r\nRequest-Id: 73cfd642-4426-11e7-8004-000000000000\r\nX-Influxdb-Version: 1.2.4\r\nDate: Mon, 29 May 2017 04:22:40 GMT\r\nTransfer-Encoding: chunked\r\n\r\n93\r\n{"results":[{"statement_id":0,"series":[{"name":"test_measurement","columns":["time","last"],"values":[["2017-05-18T21:14:56.62600105Z",110]]}]}]}\n\r\n0\r\n\r\n\r\n\r\nOK\r\n\r\nNO CARRIER\r\n'
```

**Comment by mdbartos** [comment]
Addressed in commit b02ce12057c0355033dc933e27b263c01799cca3
